middleware/xatmi/source/xatmi/internal/code.cpp

```diff
@@ -28,6 +29,9 @@ namespace casual
                if( code == common::code::casual::invalid_argument)
                   return common::code::xatmi::argument;
 
+               if( common::code::is::category< common::code::signal>( code))
+                  return common::code::xatmi::signal;
+
```

Comment:
We can't get a `code::casual::interrupted` here?

Reply:
From what I've been able to tell, where we might get `code::casual::interrupted`, we check for it and then call `signal::dispatch()` to handle the signal or raise a `code::signal`. Though there might be something I'm not seeing.